core/embed/rust/src/ui/model_mercury/component/hold_to_confirm.rs

```diff
@@ -0,0 +1,310 @@
+}
+
+#[derive(Clone)]
+enum DismissType {
+    Tap,
+    Hold,
+}
+
+impl HoldToConfirm {
+    pub fn new() -> Self {
+        let button = Button::new(ButtonContent::Empty)
+            .styled(theme::button_default())
+            .with_long_press(Duration::from_millis(2200))
+            .without_haptics();
+        Self {
+            title: Label::new(
+                "Continue\nHolding".into(),
+                Alignment::Start,
+                theme::label_title_main(),
+            )
+            .vertically_centered(),
+            area: Rect::zero(),
+            circle_color: theme::GREEN,
+            circle_pad_color: theme::GREY_EXTRA_DARK,
+            circle_inner_color: theme::GREEN_LIGHT,
+            button,
+            anim: HoldToConfirmAnim::default(),
+        }
+    }
+}
+
+impl Component for HoldToConfirm {
+    type Msg = ();
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        self.area = bounds;
+        self.button.place(Rect::snap(
+            self.area.center(),
+            Offset::uniform(120),
+            Alignment2D::CENTER,
+        ));
+        self.title.place(bounds.split_top(42).0);
+        bounds
+    }
+
+    fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
+        let btn_msg = self.button.event(ctx, event);
+        match btn_msg {
+            Some(ButtonMsg::Pressed) => {
+                self.anim.start();
+                ctx.request_anim_frame();
+                ctx.request_paint();
+            }
+            Some(ButtonMsg::Released) => {
+                self.anim.reset();
+                ctx.request_anim_frame();
+                ctx.request_paint();
+            }
+            Some(ButtonMsg::Clicked) => {
+                self.anim.reset();
+                ctx.request_anim_frame();
+                ctx.request_paint();
+            }
+            Some(ButtonMsg::LongPressed) => {
+                #[cfg(feature = "haptic")]
+                haptic::play(HapticEffect::HoldToConfirm);
+                return Some(());
+            }
+            _ => (),
+        }
+        if self.anim.is_active() {
+            ctx.request_anim_frame();
+            ctx.request_paint();
+        }
+        None
+    }
+
+    fn paint(&mut self) {
+        unimplemented!()
+    }
+
+    fn render<'s>(&'s self, target: &mut impl Renderer<'s>) {
+        let (o1, o2, o3, o4, c1, h1, r1, hap) = self.anim.eval();
+
+        let parent_cover: u8 = u8::lerp(0, 255, o1);
+        let title_alpha_start: u8 = u8::lerp(0, 255, o2);
+        let title_alpha_end: u8 = u8::lerp(0, 255, o4);
+        let circle_alpha: u8 = u8::lerp(0, 255, o3);
+        let h1 = i16::lerp(266, 0, h1);
+        let r1 = i16::lerp(0, 100, r1);
+
+        shape::Bar::new(self.area)
+            .with_fg(theme::BLACK)
+            .with_bg(theme::BLACK)
+            .with_alpha(parent_cover)
+            .render(target);
+
+        let center = self.area.center() + Offset::y(9);
```

Comment:
Lots of magic numbers in here. E.g. `9` here. Might be difficult to tune later on. 
I don't insist but if you have some good names, refactor please.

Reply:
tried to get rid of some https://github.com/trezor/trezor-firmware/pull/3776/commits/834ede0d3f37cd4d9d10e1d5a89626cef78385b6